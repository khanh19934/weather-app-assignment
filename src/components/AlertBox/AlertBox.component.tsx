import React, { ReactElement } from 'react';

interface IProps {
  severity: 'warning' | 'error';
  children: string | ReactElement;
}
const AlertBox: React.FC<IProps> = ({ severity, children }: IProps) => {
  return (
    <div className={`alert ${severity === 'warning' ? 'alert-warning' : 'alert-danger'}`}>
      {children}
    </div>
  );
};

export default AlertBox;
