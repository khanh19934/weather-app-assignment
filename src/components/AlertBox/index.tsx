export { default } from './AlertBox.component';
