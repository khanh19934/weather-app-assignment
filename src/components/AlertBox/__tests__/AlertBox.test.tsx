import React from 'react';
import renderer from 'react-test-renderer';
import AlertBox from '../AlertBox.component';

it('render correctly', () => {
  const tree = renderer
    .create(
      <AlertBox severity='warning'>
        For get exactly your current location because of blocked location permission. Please enable
        location.
      </AlertBox>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('render correctly with severity error', () => {
  const tree = renderer
    .create(<AlertBox severity='error'>Sorry we can not find city name abc</AlertBox>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
