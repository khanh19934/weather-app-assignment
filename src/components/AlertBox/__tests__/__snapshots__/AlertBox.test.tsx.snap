// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="alert alert-warning"
>
  For get exactly your current location because of blocked location permission. Please enable location.
</div>
`;

exports[`render correctly with severity error 1`] = `
<div
  className="alert alert-danger"
>
  Sorry we can not find city name abc
</div>
`;
